DOCS: Adding abbreviations

mathiasayivor opened the pull request:
<!--What issue does this pull request close -->

Closes #92 

### What new changes did you make? Tick all applicable boxes
- [ ] Added new abbreviation
- [x] Fixed something in the source code
- [x] Added a new feature
- [x] Fixed the docs (README.md, CONTRIBTUING.md, etc)

<!--Give a brief outline of changes you made. If you added slangs, which ones? -->
### Describe the new changes you added.
* Added the docs for adding a new slang
* Created a CLI to make adding new slangs easy

<!--Optional, but advised -->
### Share a screeshot of new changes

mathiasayivor commented:
@Njong392 @adropofilm kindly review the changes.

Note: I did really poor documentation on using the CLI (the CLI is very easy to use though), so modifications are welcomed.
Also, I just realised the `structure` branch is not yet merged. But it has to be merged first before merging this PR since this is a continuation.

Njong392 commented:
I like the new additions. Maybe we can move the cli and other documentation you have added over to the main README file. I just merged the `structure` branch. Also, about the cli method. A similar pull request was opened (check #106 ) Perhaps you could  take a look at that.

mathiasayivor commented:
> I like the new additions. Maybe we can move the cli and other documentation you have added over to the main README file. I just merged the `structure` branch. Also, about the cli method. A similar pull request was opened (check #106 ) Perhaps you could take a look at that.

Sure. Would append the new changes to the main README file.
I just went through #106 and observed it's not compatible with the new changes in the DB structure.

adropofilm commented:
So moving forward, do we want to merge this without the CLI documentation and have the guy who opened #106 refactor it? @Njong392 @mathiasayivor

Njong392 commented:
> So moving forward, do we want to merge this without the CLI documentation and have the guy who opened #106 refactor his changes? @Njong392 @mathiasayivor

I have commented on the issue and told the author to check out the new conversation here. Let's see what they might have to add.

mathiasayivor commented:
> So moving forward, do we want to merge this without the CLI documentation and have the guy who opened #106 refactor his changes? @Njong392 @mathiasayivor

Well, that depends.
If the goal is to re-create already existing solutions, then it makes sense. Otherwise, I suggest a review be made and changes from both solutions are considered, then combined.

Abdulmumin1 commented:
i think this solution is awesome and alot more well implemented, the only thing i want to add is, users don't have to follow the strict convention of entering slang in lower case and definitions in title case, i think that can be handled in the code without any problems.

what do you think?

mathiasayivor commented:
> i think this solution is awesome and alot more well implemented, the only thing i want to add is, users don't have to follow the strict convention of entering slang in lower case and definitions in title case, i think that can be handled in the code without any problems.
> 
> what do you think?

I agree with you.
The slang itself can be automatically converted to lowercase, but we still need to look into situations where it's not a necessity to capitalize the definitions

Njong392 commented:
I'm sorry, @mathiasayivor could you please explain the new changes you've made? The whole thing is running faster than me right now :laughing: Sorry if I'm slow in getting back to you.

mathiasayivor commented:
> I'm sorry, @mathiasayivor could you please explain the new changes you've made? The whole thing is running faster than me right now laughing Sorry if I'm slow in getting back to you.

It's okay @Njong392 😅 

> > I like the new additions. Maybe we can move the cli and other documentation you have added over to the main README file. I just merged the `structure` branch. Also, about the cli method. A similar pull request was opened (check #106 ) Perhaps you could take a look at that.
> 
> Sure. Would append the new changes to the main README file. I just went through #106 and observed it's not compatible with the new changes in the DB structure.

In summary, the changes I've made after the above discussion
* Moved instructions for adding new abbreviations using the CLI to the main README
* CLI now automatically converts abbreviations to lowercase letters as suggested by @Abdulmumin1 
* Updated the README in `public/server` to only include instructions for exporting abbreviations in db.json, cleaning up exported abbreviations in the `public/server/db` directory and removing individual abbreviations, all using the CLI.
* Added confirmation message to db cleanup CLI to prevent users from accidentally cleaning up exported abbreviations

adropofilm commented:
Thanks for the explanation @mathiasayivor, the improvements are really useful! LGTM! @Njong392 any thoughts?

Njong392 commented:
@mathiasayivor these are some HUGE changes. Like WOW haha It's amazing. Why the `cleanup` functionality though? I'm still trying to wrap my head around that.

mathiasayivor commented:
> Why the `cleanup` functionality though? I'm still trying to wrap my head around that.

It becomes handy when you want to clear (remove) all abbreviations in the `public/server/db` directory.

Njong392 commented:
> > Why the `cleanup` functionality though? I'm still trying to wrap my head around that.
> 
> It becomes handy when you want to clear (remove) all abbreviations in the `public/server/db` directory.

I see. All looks good to me :+1: As a random follow up question, do you have a twitter account by any chance?

mathiasayivor commented:
> > > Why the `cleanup` functionality though? I'm still trying to wrap my head around that.
> > 
> > 
> > It becomes handy when you want to clear (remove) all abbreviations in the `public/server/db` directory.
> 
> I see. All looks good to me +1 As a random follow up question, do you have a twitter account by any chance?

yep @mats_nk
